modules/javafx.controls/src/test/java/test/javafx/scene/control/AcceleratorParameterizedTest.java

```diff
@@ -297,4 +300,77 @@ private void assertSceneDoesNotContainKeyCombination(KeyCombination keyCombinati
         keyboard.doKeyPress(KeyCode.DIGIT1, KeyModifier.ALT);
         assertEquals(0, eventCounter);
     }
+
+    @Test public void testAcceleratorShouldNotGetFiredWhenMenuItemRemovedFromScene() {
+        KeyEventFirer kb = new KeyEventFirer(item1, scene);
+
+        kb.doKeyPress(KeyCode.DIGIT1, KeyModifier.ALT);
+        assertEquals(1, eventCounter);
+        assertEquals(1, getListenerCount(item1.acceleratorProperty()));
+
+        // Remove all children from the scene
+        Group root = (Group)scene.getRoot();
+        root.getChildren().clear();
+
+        assertEquals(0, getListenerCount(item1.acceleratorProperty()));
+        kb.doKeyPress(KeyCode.DIGIT1, KeyModifier.ALT);
+        assertEquals(1, eventCounter);
+    }
+
+    @Test public void testAcceleratorShouldGetFiredWhenMenuItemRemovedAndAddedBackToScene() {
+        KeyEventFirer kb = new KeyEventFirer(item1, scene);
+
+        kb.doKeyPress(KeyCode.DIGIT1, KeyModifier.ALT);
+        assertEquals(1, eventCounter);
+
+        Group root = (Group)scene.getRoot();
+        scene.setRoot(new Group()); // Remove all children from the scene
+        scene.setRoot(root); // Add the children to the same scene
+
+        assertEquals(1, getListenerCount(item1.acceleratorProperty()));
+        kb.doKeyPress(KeyCode.DIGIT1, KeyModifier.ALT);
+        assertEquals(2, eventCounter);
+    }
+
+    @Test public void testAcceleratorShouldGetFiredWhenMenuItemRemovedAndAddedToDifferentScene() {
+        KeyEventFirer kb = new KeyEventFirer(item1, scene);
+
+        kb.doKeyPress(KeyCode.DIGIT1, KeyModifier.ALT);
+        assertEquals(1, eventCounter);
+
+        Group root = (Group)scene.getRoot();
+        scene.setRoot(new Group()); // Remove all children from the scene
+        Scene diffScene = new Scene(root); // Add the children to a different scene
+        sl.getStage().setScene(diffScene);
+        kb = new KeyEventFirer(item1, diffScene);
+
+        assertEquals(1, getListenerCount(item1.acceleratorProperty()));
+        kb.doKeyPress(KeyCode.DIGIT1, KeyModifier.ALT);
+        assertEquals(2, eventCounter);
+    }
+
+    @Ignore("Passes only for Button")
```

Comment:
Can you file the follow-on bug and then update this comment to use that bug ID?

Reply:
Updated with bug ID [JDK-8268374](https://bugs.openjdk.java.net/browse/JDK-8268374)